docs/core-concepts/core-sdk/cli/dotnet-build.md

```diff
@@ -0,0 +1,87 @@
+dotnet-build
+===========
+
+## NAME 
+dotnet-build -- builds a project and all of its' dependencies 
+
+## SYNOPSIS
+
+dotnet-build [--output]  
+    [--build-base-path] [--framework]  
+    [--configuration]  [--runtime] [--version-suffix]
+    [--build-profile]  [--no-incremental] [--no-dependencies]
+    [< project >]  
+
+## DESCRIPTION
+
+`dotnet-build` builds multiple source file from a source project and its dependencies into a binary. 
+The binary will be in Intermmidiate Language (IL) by default and will have a DLL extension. 
+`dotnet-build` will also drop a \*.deps file which outlines what the runner needs to run the application.  
+
+Building requires an existence of a lock file which means that a `dotnet-restore` call needs to happen 
+previous to building.
+
+Before any compilation begins, the build verb analyzes the project and its dependencies for incremental safety checks. 
+If all checks clear out, then build proceeds with incremental compilation of the project and its dependencies; 
+otherwise it falls back to non-incremental compilation. Via a profile flag, users can choose to receive additional 
+information on how they can improve their build times.
+
+All the projects in the dependency graph that need compilation must pass the following safety checks in order for the 
+compilation process to be incremental:
+- not use pre / post compile scripts
+- not load compilation tools from PATH (e.g., resgen, compilers)
+- use only known compilers (csc, vbc, fsc)
+
+In order to build an executable application (console application), you need a special configuration section in project.json:
+
+```json
+{ 
+    "compilerOptions": {
+      "emitEntryPoint": true
+    }
+}
+```
+
+Class libraries do not need this special piece of configuration. 
+
```

Comment:
I think this can be simplified. The CLI can only output two types of binaries: exe an dll. You don't need to call out the bit about console applications, or that class libraries do not need this. (Again, instructing the user that they don't need to do something is strange to me.)